### PR TITLE
Mount all unmounted fsses under '$fs$/fs<idx>'

### PR DESCRIPTION
**Pull request by Miauwkeru**
(DIS-2250)

### CODE REVIEW

File: dissect/target/target.py [hunk 1]
```diff
@@ -481,6 +482,14 @@ def _init_os(self) -> None:
         self._os_plugin = os_plugin
         self._os = self.add_plugin(os_plugin.create(self, fs))
 
+    def _mount_others(self) -> None:
+        root_fs = self.fs
+        index = 0
+        for fs in self.filesystems:
+            if fs not in root_fs.mounts.values():
+                root_fs.mount(f"/$fs$/fs{index}", fs)
```
Comment (Schamper) [inline dissect/target/target.py:490]: Can you add a check for if `$fs$` already exists?

**Review by Schamper — CHANGES_REQUESTED** [review]

File: dissect/target/target.py (hunk 1 above)
Comment (pyrco) [reply dissect/target/target.py:494]: Or maybe under a different index? So you can re-run this function if you've added new filesystems.

File: dissect/target/target.py [hunk 2]
```diff
@@ -481,6 +482,22 @@ def _init_os(self) -> None:
...
+    def _mount_others(self) -> None:
+        root_fs = self.fs
+        index = 0
+
+        # determine mount point directory
+        path = "/$fs$"
+        counter = 0
+        while root_fs.path(path).exists():
+            path = f"/$fs_{counter}$"
+            counter += 1
```
Comment (pyrco) [inline dissect/target/target.py:494]: I was more thinkin along the lines of
```
path = f"/$fs$/fs{counter}"
```

**Review by pyrco — CHANGES_REQUESTED** [review]

File: dissect/target/target.py [hunk 3]
```diff
@@ -482,6 +483,20 @@ def _init_os(self) -> None:
...
+    def _mount_others(self) -> None:
+        root_fs = self.fs
+        counter = 0
+        path = "/$fs$/fs0"
+
+        for fs in self.filesystems:
+            # determine mount point
+            while root_fs.path(path).exists():
+                counter += 1
+                path = f"/$fs$/fs{counter}"
```
Comment (pyrco) [inline dissect/target/target.py:495]: Wouldn't it be more logical/efficient to have this while loop inside the `if` below?

**Review by pyrco — CHANGES_REQUESTED** [review]

File: tests/test_target.py [hunk 4]
```diff
@@ -499,3 +499,27 @@ def test_vs_offset_0(target_bare: Target) -> None:
         filesystem_open.assert_called_once_with(mock_volume)
         assert len(target_bare.volumes) == 1
         assert len(target_bare.filesystems) == 1
+
+
+@pytest.mark.parametrize("nr_of_fs", [1, 2])
+def test_fs_mount_others(target_unix: Target, nr_of_fs: int):
```
Comment (Schamper) [inline tests/test_target.py:505]:
```suggestion
def test_fs_mount_others(target_unix: Target, nr_of_fs: int) -> None:
```

File: tests/test_target.py [hunk 5]
```diff
@@ -499,3 +499,27 @@ def test_vs_offset_0(target_bare: Target) -> None:
...
+
+    for idx in range(nr_of_fs):
+        assert f"/$fs$/fs{idx}" in target_unix.fs.mounts.keys()
+        assert target_unix.fs.path(f"$fs$/fs{idx}").exists()
+
+    assert not target_unix.fs.path(f"$fs$/fs{nr_of_fs}").exists()
+
+
+@pytest.mark.parametrize("nr_of_fs", [1, 2])
+def test_fs_mount_already_there(target_unix: Target, nr_of_fs: int):
```
Comment (Schamper) [inline tests/test_target.py:519]:
```suggestion
def test_fs_mount_already_there(target_unix: Target, nr_of_fs: int) -> None:
```

**Review by Schamper — CHANGES_REQUESTED** [review]

**Review by pyrco — APPROVED** [review]

**Pull request merged**